app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/DefaultSampleInfoRepository.kt

```diff
@@ -36,62 +35,61 @@ import kotlinx.serialization.json.Json
  */
 object DefaultSampleInfoRepository : SampleInfoRepository {
 
-    private val sampleList = mutableListOf<Sample>()
-    private val sampleListMutex = Mutex()
+    private val isInitialized = AtomicBoolean(false)
 
-    private suspend fun updateSampleList(block: MutableList<Sample>.() -> Unit) {
-        sampleListMutex.withLock { sampleList.block() }
-    }
+    private val sampleList = mutableListOf<Sample>()
 
     /**
      * Load the sample metadata from the metadata folder in the assets directory and updates sampleList
      * of [Sample] objects.
      */
     suspend fun load(context: Context) {
-        // Iterate through the metadata folder for all metadata files
-        val json = Json { ignoreUnknownKeys = true }
-        context.assets.list("samples")?.forEach { samplePath ->
-            // Get this metadata files as a string
-            context.assets.open("samples/$samplePath/README.metadata.json").use { inputStream ->
-                val metadataJsonString = inputStream.bufferedReader().use { it.readText() }
-                try {
-                    val metadata = json.decodeFromString<SampleMetadata>(metadataJsonString)
+        if (isInitialized.compareAndSet(false, true)) {
+            // Iterate through the metadata folder for all metadata files
+            val json = Json { ignoreUnknownKeys = true }
```

Comment:
This should be initialized as a property outside of the function because it is an expensive call and thus a simple performance improvement.